### DISCUSSION TITLE
From 1.0.0-next.286 how to correctly type the Load function?

### DISCUSSION THREAD

**Discussion by frederikhors** [sveltejs/kit#4141] [Q&A]
After upgrade from `1.0.0-next.285` to `1.0.0-next.286` I have an error using this code:

```svelte
<script context="module" lang="ts">
	import type { LoadInput, LoadOutput } from "@sveltejs/kit";

	export function load({ url }: LoadInput): LoadOutput {
		//...
	}
</script>
```

Now if I copy the code from https://kit.svelte.dev/docs/loading:

```svelte
<script context="module" lang="ts">
  /** @type {import('@sveltejs/kit').Load} */
  export function load({ url }) {
    //...
  }
</script>
```

the `url` is `any`.

Why?

Am I wrong using it?

**Comment by mustofa-id** [comment]
There is [breaking](https://github.com/sveltejs/kit/pull/4104) changes about type on [1.0.0-next.286](https://github.com/sveltejs/kit/blob/master/packages/kit/CHANGELOG.md#100-next286). The `LoadInput` and `LoadOutput` now exported from `@sveltejs/kit/types/private`. So the correct code should be:

```svelte
<script context="module" lang="ts">
	import type { LoadInput, LoadOutput } from "@sveltejs/kit/types/private";

	export function load({ url }: LoadInput): LoadOutput {
		//...
	}
</script>
```

I prefer using function expression because it simply import only `Load` type.

```svelte
<script lang="ts" context="module">
	import type { Load } from '@sveltejs/kit'

	export const load: Load = async ({ url }) => {
		// url === URL
		return {}
	}
</script>

```

**Reply by frederikhors** [reply]
So the code `/** @type {import('@sveltejs/kit').ErrorLoad} */` is not for Typescript?

**Reply by frederikhors** [reply]
Using the code you propose I cannot use `return { props: { status } };` in __error.svelte.

**Reply by mustofa-id** [reply]
The `@type` in comment is [JSDoc](https://jsdoc.app/) tag that provide type information for JavaScript code.
In [custom error page](https://kit.svelte.dev/docs/layouts#error-pages), make sure you are using `ErrorLoad` type instead of `Load` and or `ErrorLoadInput` instead of `LoadInput`.

**Reply by frederikhors** [reply]
I tried with:

```svelte
<script context="module" lang="ts">
	import type { ErrorLoad } from "@sveltejs/kit";

	export function load({ status }): ErrorLoad {
		return { props: { status } };
	}
</script>
```

I get this error:

```
Type '{ props: { status: any; }; }' is not assignable to type 'ErrorLoad<Record<string, string>, Record<string, any>>'.
  Object literal may only specify known properties, and 'props' does not exist in type 'ErrorLoad<Record<string, string>, Record<string, any>>'.
```

**Reply by mustofa-id** [reply]
It should be:
```ts
export const load: ErrorLoad = ({ status }) => {
	return { props: { status } }
}
```
or if you prefer function declaration:
```ts
export function load({ status }: ErrorLoadInput): LoadOutput {
	return { props: { status } }
}
```

**Reply by frederikhors** [reply]
> or if you prefer function declaration:

My question here is about this indeed! 😄

**Reply by mustofa-id** [reply]
Ah sorry. I mean you cannot use `ErrorLoad` if you use function declaration.

**Reply by frederikhors** [reply]
But why is it changed?

**Reply by mustofa-id** [reply]
There is a reason in the link in my first answer.
